modules/javafx.graphics/src/main/java/javafx/scene/Scene.java

```diff
@@ -1346,12 +1359,33 @@ static WritableImage doSnapshot(Scene scene,
             context.camera = null;
         }
 
-        // Grab the lights from the scene
+        // Grab the lights from the scene and/or subscene
         context.lights = null;
-        if (scene != null && !scene.lights.isEmpty()) {
-            context.lights = new NGLightBase[scene.lights.size()];
-            for (int i = 0; i < scene.lights.size(); i++) {
-                context.lights[i] = scene.lights.get(i).getPeer();
+        int totalLightCount = 0;
```

Comment:
Here's a suggestion: you might be able to replace all of the new code, including the `accumulateLightsForSnapshot` method, with the following shorter code:

```java
Stream<NGLightBase> lights = Stream.concat(
    Optional.ofNullable(scene).stream().flatMap(s -> s.lights.stream()).map(LightBase::getPeer),
    Optional.ofNullable(subScene).stream().flatMap(s -> s.getLights().stream()).map(LightBase::getPeer));

context.lights = lights.toArray(NGLightBase[]::new);
```